react/week3/todos/src/components/todolist.js

```diff
@@ -0,0 +1,180 @@
+            </label>
+            <input type='submit' value='Add todo' />
+          </form>
+        </div>
+      )
+    }
+}
+
+function FancyBorder(props) {
+  return (
+    <div className={'FancyBorder'}>
+      {props.children}
+    </div>
+  )
+}
+
+class Todo extends React.Component {
+    state = {
+      isEditing: false
+    }
+  
+    toggleState = () => {
+      this.setState({
+        isEditing: !this.state.isEditing
+      })
+    }
+  
+    updateTodo = (event) => {
+      event.preventDefault()
+      this.props.editTodo(this.props.index, this.input.value);
+      this.toggleState()
+    }
+  
+    renderEditMode = () => {
+      return (
+        <li>
+          <FancyBorder>
+            <form onSubmit={this.updateTodo}>
+              <input type='text' ref={(value) => { this.input = value }} defaultValue={this.props.todo.description} required /> | {this.props.todo.deadline}
+              <input type="checkbox" checked={this.props.todo.checked} onChange={() => {this.props.handleCheckbox(this.props.index)}}/>
+              <button onClick={() => {this.props.deleteTodo(this.props.index)}}>Delete</button>
+              <button type='submit'>Update</button>
+            </form>
+          </FancyBorder>      
+        </li>      
+      )
+    }
+  
+    renderTodoList = () => {
+      return (
+        <li style={{textDecoration: this.props.todo.checked ? 'line-through' : 'none'}}>
+          <FancyBorder>
+            <label>{this.props.todo.description} | {this.props.todo.deadline}</label>
+            <input type="checkbox" checked={!!this.props.todo.checked} onChange={() => {this.props.handleCheckbox(this.props.index)}}/>
+            <button onClick={() => {this.props.deleteTodo(this.props.index)}}>Delete</button>
+            <button onClick={(event) => { 
+              event.stopPropagation();
+              this.toggleState()
+              }}>Edit</button>
+          </FancyBorder>       
+        </li> 
+      )
+    }
+    render() {
+      return (
+        <div>
+          { this.state.isEditing ? this.renderEditMode() : this.renderTodoList() }
+        </div>
+      )
+    }
+  }
+  
+  function fetchTodo() {
+    const URL = `https://gist.githubusercontent.com/benna100/391eee7a119b50bd2c5960ab51622532/raw`
+    return fetch(URL).then(response => response.json())
+  }
+  
+  
+
+class TodoList extends React.Component {
+    state = {
+        description: '',
+        deadline: '',
+        checked: false,
+        todoList: []
+      }
+  
+    componentDidMount() {
+      this.fetchTodos()
+    }
+  
+    fetchTodos = () => {
+      fetchTodo().then(todo => {
+        this.setState({ todoList: todo})
+      })
+    }
+  
+    handleSubmit = (event) => {   
+      this.setState({
+        description: this.state.description,
+        deadline: this.state.deadline,
+        todoList: [...this.state.todoList, {description: this.state.description, deadline: this.state.deadline, checked:false} ]
+      })
```

Comment:
```suggestion
      this.setState({
        todoList: [...this.state.todoList, {description: this.state.description, deadline: this.state.deadline, checked:false} ]
      })
```

You don't need to pass everything along when you update state, it's ok to only update the properties you're interested in updating.